capa/features/extractors/base_extractor.py

```diff
@@ -296,6 +296,37 @@ def extract_insn_features(
         raise NotImplementedError()
 
 
+class StaticFeatureExtractorFilter:
+    def __init__(self, inner: StaticFeatureExtractor):
+        self.inner = inner
+
+    def __getattr__(self, attr):
+        if attr in self.__dict__:
+            return getattr(self, attr)
+        return getattr(self.inner, attr)
+
+    @property
+    def __class__(self):
+        return self.inner.__class__
+
+    @__class__.setter
+    def __class__(self, value) -> None:
+        self.inner.__class__ = value
+
+    @classmethod
+    def __instancecheck__(cls, instance):
+        return isinstance(instance, StaticFeatureExtractor)
```

Comment:
I have reverted to the previous implementation I had (modifying the extractor's functions). However, I tried to copy the extractor before I alter the method (essentially changing the method of the copy of the extractor object and returning that new copy). This gets around the issue we had before (tests not working properly since filters would alter the extractor object).

Another approach that might be good as well is to add a list of processes or functions to filter to the static or dynamic feature extractor abstract class, and then updating that list with each call to `ProcessFilter()` or `FunctionFilter()` respectively, but I am not sure if that's better than the current approach.

Let me know if you think that approach would be better (or if you have any other ideas!).